### DISCUSSION TITLE
Rendering and optimisation of videos

### DISCUSSION THREAD

**Discussion by SebastianNegus** [mozilla/hubs#3022] [General Questions & Discussion] [answered]
Hi,

I am currently working on an exhibition with around 20 playing videos, 20 PDFs and posters. I am trying to see how to best optimise the layout of the room in order to accommodate 22 users. My main concern is about the performance of the room because there are so many videos. We are going to host the files on Amazon S3.
I was wondering how does the rendering work in hubs? So i am seeing that when i look towards a lump of videos the frames per second go down, which is understandable. But I am surprised that still happens even if there is a wall in-between the character and the videos. Is there a way of stopping that from happening so the performance doesn't drop when just randomly looking around?
How would you go about creating something with these requirements?

Thanks

**Comment by robinkwilson** [comment, marked as answer]
I suggest creating separate rooms to break up these videos -- your users can move between the rooms for each one. We loosely recommend 3 videos per room for the best performance.

**Reply by emclaren** [reply]
another work-around (if your video doesn't have sound) we've seen folks use is combining multiple videos into one screen (like the Brady Bunch intro video on TV - for an example check out the video wall behind the apart poster team here: https://www.youtube.com/watch?v=oMkNOAl8y6A, note that sometimes the video looks a bit frozen this was due to issues with my screencapturing it looked fine in-world)
